### DISCUSSION TITLE
Blog comment on Verba: Building an Open Source, Modular RAG Application

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#1834] [Docs/blog comments]
# Blog comment on Verba: Building an Open Source, Modular RAG Application

Verba is an open source Retrieval Augmented Generation (RAG) application built using a modular, customizable architecture that makes it easy for anyone to use AI methods to get personalized answers on their own data. 

https://weaviate.io/blog/verba-open-source-rag-app

**Comment by Ayesha-Imr** [comment]
Verba sounds fantastic! I can’t wait to try it out and play around with it. Adding multi-tenancy to Verba would be a great feature as it provides a lot of benefits especially regarding privacy concerns.

**Comment by geeman11** [comment]
Thank you for making Verba guys! I was messing around with RAG for a very, very long time and never got the results i wanted. I originally used Verba to scour the Magic The Gathering rulebook to quickly put those table debates to rest (with evidence). I showed my boss the power of RAG and was quickly asked if it could search through resume's to find people worth interviewing... it worked quite well! Theres still a lot of testing i need to do with choosing embedding models, generation models, and refining the configuration, but that is normal! Thank you again for making RAG a lot more accessible.
